### PR TITLE
Add grisp pack command

### PR DESCRIPTION
**Pull request by sylane**
Require: https://github.com/grisp/grisp_tools/pull/28

### CODE REVIEW

**Comment by sylane** [comment]
The dependency will be fix when grisp_tools is published

**Comment by sylane** [comment]
I just saw I forgot to change and uncomment the usage got the pack command, I'll do it tomorrow...

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - New `-t/--tar` option to the deploy command to save a grisp release tarball in
   the `_grisp/deploy` directory.
 - New firmware command to generate GRiSP 2 binary firmwares: [#83](https://github.com/grisp/rebar3_grisp/pull/83)
+- New pack command to generate GRiSP 2 software update package: [#84](https://github.com/grisp/rebar3_grisp/pull/84)
```
Comment (ziopio) [inline CHANGELOG.md:15]: wrong link should be PR 85

File: src/rebar3_grisp_pack.erl [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+            <<"HOSTNAME">>
+    end.
+
+update_usage(RState, RelName0, RelVsn) ->
+    RelName = atom_to_binary(RelName0),
+    PackageFile0 = rebar3_grisp_util:update_file_path(RState, RelName, RelVsn),
+    PackageFile = grisp_tools_util:maybe_relative(PackageFile0, ?MAX_DDOT),
+    iolist_to_binary([
+        "Instructions to update GRiSP2 software on the board with grisp_updater:\n",
+        "     NOTE: the software running on the board MUST run grisp_updater by adding the grisp_updater_grisp2 dependency.\n",
```
Comment (ziopio) [inline src/rebar3_grisp_pack.erl:281]: We need to publish grisp_updater_grisp2 on github, and explain this in the README at least, we could also move advanced topics in a separate markdown to reduce the clutter to first time visitors of the github repo.

File: src/rebar3_grisp_pack.erl [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+            ]},
+            {profiles, [grisp]},
+            {short_desc, "Generate a GRiSP software update package"},
+            {desc,
+                "Generate a tarball that can be used with grisp software update.\n"
+                "\n"
+                "If no firmware file is specified, it will be generated by "
+                "calling 'rebar3 grisp firmware' with the optional release name "
+                "and version. As for the firmware command, options passed after "
+                "'--' are sent to the rebar 3 release task.\n"
```
Comment (ziopio) [inline src/rebar3_grisp_pack.erl:52]: As last thing we should document the purpose of this command in a markdown file at least, I would move it in ADVANCED.md toghether with the formware command explanation to remove extra clutter from the main README. I would just mention these and link the secondary document.

File: src/rebar3_grisp_pack.erl [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+            {name, pack},
+            {module, ?MODULE},
+            {bare, true},
+            {deps, [{default, install_deps}, {default, compile}]},
+            {example, "rebar3 grisp pack"},
+            {opts, [
+                {relname, $n, "relname", string, "Specify the name for the release that will be deployed"},
+                {relvsn, $v, "relvsn", string, "Specify the version of the release"},
+                {system, undefined, "system", string, "An explicit system firmware to use in the update package"},
+                {bootloader, undefined, "bootloader", string, "An explicit bootloader firmware to use in the update package"},
```
Comment (ziopio) [inline src/rebar3_grisp_pack.erl:37]: TODO make bootloader and toolchain usage optional

**Review by ziopio — CHANGES_REQUESTED** [review]

File: src/rebar3_grisp_pack.erl [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+            <<"HOSTNAME">>
+    end.
+
+update_usage(RState, RelName0, RelVsn) ->
+    RelName = atom_to_binary(RelName0),
+    PackageFile0 = rebar3_grisp_util:update_file_path(RState, RelName, RelVsn),
+    PackageFile = grisp_tools_util:maybe_relative(PackageFile0, ?MAX_DDOT),
+    iolist_to_binary([
+        "Instructions to update GRiSP2 software on the board with grisp_updater:\n",
+        "     NOTE: the software running on the board MUST run grisp_updater by adding the grisp_updater_grisp2 dependency.\n",
```
Comment (GwendalLaurent) [inline src/rebar3_grisp_pack.erl:320]: On a side note, we should probably add the option to include `grisp_updater` when we run `rebar3 grisp configure`. But this an additional feature and can be done in a future PR

File: src/rebar3_grisp_pack.erl (hunk 2 above)
Comment (sylane) [reply src/rebar3_grisp_pack.erl:320]: It is explained in the README.md now.

File: src/rebar3_grisp_pack.erl (hunk 3 above)
Comment (sylane) [reply src/rebar3_grisp_pack.erl:53]: Yes, a wiki tutorial would be nice, and some documentation about A/B software update in general. Not part of this PR though.

File: src/rebar3_grisp_pack.erl [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+            {example, "rebar3 grisp pack"},
+            {opts, [
+                {relname, $n, "relname", string, "Specify the name for the release that will be deployed"},
+                {relvsn, $v, "relvsn", string, "Specify the version of the release"},
+                {system, undefined, "system", string, "An explicit system firmware to use in the update package"},
+                {bootloader, undefined, "bootloader", string, "An explicit bootloader firmware to use in the update package"},
+                {block_size, undefined, "block-size", integer, "The size of the blocks in bytes, before compression"},
+                {key, $k, "key", string, "Private key PEM file to use to sign the update package"},
+                {force_firmware, $F, "force-firmware", {boolean, false}, "Force firmware building even if it already exists"},
+                {force, $f, "force", {boolean, false}, "Replace existing files"},
```
[the remark this answers is not in the document]
Comment (sylane) [reply src/rebar3_grisp_pack.erl:42]: It was at first, but then it mean that if the file exists you are forces to generate a firmware...
I could change it back to have only one force option, in this command and in firmware too, that is similar.

**Review by ziopio — APPROVED** [review]
Good stuff

**Pull request merged**